lib/mediumClient.js

```diff
@@ -0,0 +1,411 @@
+ *  canonicalUrl: string,
+ *  publishStatus: PostPublishStatus,
+ *  license: PostLicense
+ * }} options
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.createPostInPublication = function (options, callback) {
+  this._enforce(options, ['publicationId'])
+  this._makeRequest({
+    method: 'POST',
+    path: '/v1/publications/' + options.publicationId + '/posts',
+    data: {
+      title: options.title,
+      content: options.content,
+      contentFormat: options.contentFormat,
+      tags: options.tags,
+      canonicalUrl: options.canonicalUrl,
+      publishStatus: options.publishStatus,
+      license: options.license
+    }
+  }, true, callback)
+}
+
+
+/**
+ * Acquires an access token for the Medium API.
+ *
+ * Sets the access token on the client on success.
+ *
+ * @param {Object} params
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype._acquireAccessToken = function (params, callback) {
+  this._makeRequest({
+    method: 'POST',
+    path: '/v1/tokens',
+    contentType: 'application/x-www-form-urlencoded',
+    data: qs.stringify(params)
+  }, false, function (err, data) {
+    if (!err) {
+      this._accessToken = data.access_token
+    }
+    callback(err, data)
+  }.bind(this))
+}
+
+
+/**
+ * Enforces that given options object (first param) defines
+ * all keys requested (second param). Raises an error if any
+ * is missing.
+ *
+ * @param {Object} options
+ * @param {keys} requiredKeys
+ */
+MediumClient.prototype._enforce = function (options, requiredKeys) {
+  if (!options) {
+    throw new MediumError('Parameters for this call are undefined', DEFAULT_ERROR_CODE)
+  }
+  requiredKeys.forEach(function (requiredKey) {
+    if (!options[requiredKey]) throw new MediumError('Missing required parameter "' + requiredKey + '"', DEFAULT_ERROR_CODE)
+  })
+}
+
+
+
+/**
+ * Makes a request to the Medium API.
+ *
+ * @param {Object} options
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype._makeRequest = function (options, unwrapDataEnvelope, callback) {
+  var requestParams = {
+    host: 'api.medium.com',
+    port: 443,
+    method: options.method,
+    path: options.path
+  }
+  var req = https.request(requestParams, function (res) {
+    var body = []
+
+    res.setEncoding('utf-8')
+    res.on('data', function (data) {
+      body.push(data)
+    })
+    res.on('end', function () {
+      var payload
+      var responseText = body.join('')
+      try {
+        payload = JSON.parse(responseText)
+      } catch (err) {
+        callback(new MediumError('Failed to parse response', DEFAULT_ERROR_CODE), null)
+        return
+      }
+
+      var statusCode = res.statusCode
+      var statusType = Math.floor(res.statusCode / 100)
+
+      if (statusType == 4 || statusType == 5) {
+        var err = payload.errors[0]
+        callback(new MediumError(err.message, err.code), null)
+      } else if (statusType == 2) {
+        callback(null, unwrapDataEnvelope ? payload.data : payload)
```

Comment:
fix number 2! :)

Or... you could just do `callback(null, payload.data || payload)`